ENCD-4376 ATAC-seq QC metrics

yunhailuo opened the pull request.

gabdank commented on src/encoded/schemas/atac_duplication_quality_metric.json:
```diff
@@ -0,0 +1,63 @@
+            "type": "integer"
+        },
+        "paired_opt_dupes": {
+            "title": "Paired optimal duplications",
```
I think these are optical and not "optimal"

gabdank commented on src/encoded/schemas/atac_duplication_quality_metric.json:
```diff
@@ -0,0 +1,63 @@
+            "title": "Unmapped reads",
+            "type": "integer"
+        },
+        "unpaired_dupes": {
```
here and below, I think it is "duplicates" rather than "duplications"

gabdank commented on src/encoded/schemas/atac_duplication_quality_metric.json:
```diff
@@ -0,0 +1,63 @@
+            "type": "integer"
+        },
+        "dupes_pct": {
+            "title": "Percentage duplication",
```
I think "duplicates percentage"

gabdank commented on src/encoded/schemas/atac_frip_quality_metric.json:
```diff
@@ -0,0 +1,79 @@
+        },
+        "pr1_FRiP": {
+            "title": "FRiP score of pseudoreplicate 1 peak",
+            "description": "FRiP score of pseudoreplicate 1 peak of the corresponding replicate of this peak.",
```
title: FRiP score of pseudoreplicate 1
description: Fraction of reads in the peak file of pseudoreplicate 1

gabdank commented on src/encoded/schemas/atac_frip_quality_metric.json:
```diff
@@ -0,0 +1,79 @@
+            "maximum": 1
+        },
+        "pr2_FRiP": {
+            "title": "FRiP score of pseudoreplicate 2 peak",
```
title: FRiP score of pseudoreplicate 2
description: Fraction of reads in the peak file of pseudoreplicate 2

gabdank commented on src/encoded/schemas/atac_pbc_quality_metric.json:
```diff
@@ -0,0 +1,67 @@
+            "title": "Distinct read pairs",
+            "type": "integer"
+        },
+        "one_read_pair": {
```
Does "one_read_pair" means read1 pairs? Or pair with only one read mapped?

gabdank commented on src/encoded/schemas/atac_reproducibility_quality_metric.json:
```diff
@@ -0,0 +1,110 @@
+        "schema_version": {
+            "default": "1"
+        },
+        "reproducibility_type": {
```
I think "reproducibility_type" should be required, otherwise we can end up with values without the ability to find out is it is IDR or not.

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "duplicated_reads": {
```
add minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "dedup_reads": {
```
add minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "title": "Read count after removing duplicate reads",
+            "type": "integer"
+        },
+        "mitochondrial_reads": {
```
add minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "dup_mito_reads": {
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "maximum": 1
+        },
+        "final_reads": {
+            "title": "Final reads after all filtering",
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "bowtie_stats": {
```
I am not sure what bowtie_stats output is, and if it can be negative

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "type": "integer"
+        },
+        "idr_peaks": {
+            "title": "Number of IDR peaks",
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "type": "integer"
+        },
+        "naive_overlap_peaks": {
+            "title": "Number of naive overlap peaks",
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "type": "integer"
+        },
+        "raw_peaks": {
+            "title": "Number of raw peaks",
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "description": "PCR Bottlenecking coefficient 2 (indicates library complexity) = M1/M2 where M1: number of genomic locations where only one read maps uniquely and M2: number of genomic locations where 2 reads map uniquely"
+        },
+        "picard_est_library_size": {
+            "title": "Picard estimated library size",
```
library size is an integer? minimum=0 ?

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum":0
+        },
+        "enhancer_reads": {
+            "title": "Number of reads in enhancer regions",
```
minimum=0

gabdank commented on src/encoded/schemas/ataqc_quality_metric.json:
```diff
@@ -0,0 +1,298 @@
+            "minimum":0
+        },
+        "called_peak_reads": {
+            "title": "Number of reads in called peak regions",
```
minimum=0

gabdank commented on src/encoded/schemas/atac_frip_quality_metric.json:
```diff
@@ -0,0 +1,83 @@
+            "description": "FRiP score of IDR peaks between replicate 1 and replicate 2",
+            "type": "number",
+            "minimum": 0,
+            "maximum": 1
```
if I understand correctly the "rep-rep_idr_FRiP" is going to be associated with 2 files (peaks from rep1 and rep2)? If that is the case, maybe we can give a less concrete title (since the replicates are going to be numbered differently on the portal). Something like "title": "FRiP of replicates IDR peaks", similarly "FRiP of replicates overlap peaks" ? In description, you can avoid numbering the replicates as well.

gabdank commented on src/encoded/schemas/atac_reproducibility_quality_metric.json:
```diff
@@ -0,0 +1,124 @@
+            "minimum": 0
+        },
+        "idr_cutoff":{
+            "title": "IDR cutoff",
```
Can IDR cutoff be negative?

yunhailuo replied:
Confirmed with Seth as number between 0 and 1.

gabdank commented on src/encoded/schemas/atac_xcor_score_quality_metric.json:
```diff
@@ -0,0 +1,120 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "phantom_peak": {
```
can phantom be negative?

yunhailuo replied:
Yes, it can be in terms of schema. On some plots and according to their paper, the strand shift on x-axis can start at -100bp.

gabdank commented on src/encoded/schemas/atac_xcor_score_quality_metric.json:
```diff
@@ -0,0 +1,120 @@
+            "minimum": 0,
+            "maximum": 1
+        },
+        "argmin_corr": {
```
Can argmin be negative?

yunhailuo replied:
Yes, it can be in terms of schema. On some plots and according to their paper, the strand shift on x-axis can start at -100bp.

yunhailuo commented:
Pushed on commit to address comments, @gabdank . I'll rebase once the code is fine to go.

yunhailuo commented:
Fixed a bug for pipeline modeling: adding 4 file types to input and output enum of analysis_step. Please take a look if you have time, @gabdank .

yunhailuo commented:
Changes since last approval, @gabdank 

* Removed two new file type: "optimal overlap peaks" and "conservative overlap peaks";
* Added one new file type: ""replicated idr thresholded peaks";
* Added a new property "reproducibility_classification" to File object;
* Removed "N_opt" and "N_consv" from "atac_reproducibility_quality_metric" since it will be the same as number of peaks in corresponding files and will be addressed by the new "reproducibility_classification" property;
* Removed "N_pair" and renamed "Nx" to "reproducibility_peaks" since all peak files between reps (both pseudo and true) will be posted and the number of peaks will be one per file.
* Removed FRiP scores for overlap and IDR peaks from "atac_frip_quality_metric" since it will just be "FRiP" score on corresponding peak files.
